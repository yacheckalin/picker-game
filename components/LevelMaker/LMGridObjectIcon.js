import React from "react";
import PropTypes from "prop-types";

import styled from "styled-components";
import { renderIconMapper } from "../../helpers";

const StyledIcon = styled.i`
  font-size: 2em;
`;

const LMGridObjectIcon = ({ tag, className = "" }) => {
  const { color, name } = renderIconMapper(tag);

  return (
    <StyledIcon
      className={`material-icons text-darken-2 ${color} ${className}`}
    >
      {name}
    </StyledIcon>
  );
};

LMGridObjectIcon.defaultProps = {
  className: "",
};

export default LMGridObjectIcon;
