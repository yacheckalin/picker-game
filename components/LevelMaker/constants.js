export const MIN_MAP_SIZE = 3;
export const MAX_MAP_SIZE = 50;
export const DEFAULT_GRID_MAP = [
  [0, 0, 0],
  [0, 0, 0],
  [0, 0, 0],
];

export const DEFAULT_CELL_SIZE = 40;
export const MARK_SELECTED = 999;
