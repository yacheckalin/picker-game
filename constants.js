export const KEYS = {
  BLUE_KEY: [20, "BLUE"],
  GREEN_KEY: [21, "GREEN"],
  RED_KEY: [22, "RED"],
};
export const DOORS = {
  BLUE_DOOR: [90, "BLUE"],
  GREEN_DOOR: [91, "GREEN"],
  RED_DOOR: [92, "RED"],
};
export const WALL = 1;
export const WALL_D = 2; // walls behind the door
export const VISITED = 8;
export const EMPTY = 0;
export const EMPTY_BACKPACK_CELL = 0;
export const DEFAULT_FIELD_SIZE = 4;
export const DEFAUTL_FIELD_HASH = "level_default";

export const MIN_MAP_SIZE = 3;
export const MAX_MAP_SIZE = 50;
export const DEFAULT_GRID_MAP = [
  [0, 0, 0],
  [0, 0, 0],
  [0, 0, 0],
];

export const DEFAULT_CELL_SIZE = 40;
export const MARK_SELECTED = 999;
export const MARK_CLEAR = 0;
