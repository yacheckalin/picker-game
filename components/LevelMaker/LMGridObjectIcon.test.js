import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import { DOORS, KEYS, WALL, WALL_D } from "../../constants";

import LMGridObjectIcon from "./LMGridObjectIcon";

describe("<LMGridObjectIcon />", () => {
  const initProps = {
    tag: DOORS.RED_DOOR,
  };
  let wrapper = null;
  beforeEach(() => {
    wrapper = shallow(<LMGridObjectIcon {...initProps} />);
  });
  afterEach(() => {
    wrapper.unmount();
  });
  test("renders and displays properly", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("render keys icons properly", () => {
    wrapper = shallow(<LMGridObjectIcon tag={KEYS.GREEN_KEY} />);
    expect(wrapper.contains("vpn_key")).toBe(true);
  });
  test("render doors icons properly", () => {
    wrapper = shallow(<LMGridObjectIcon tag={DOORS.GREEN_DOOR} />);
    expect(wrapper.contains("lock_outline")).toBe(true);
  });
  test("render walls icons properly", () => {
    wrapper = shallow(<LMGridObjectIcon tag={WALL} />);
    expect(wrapper.contains("border_all")).toBe(true);
  });
  test("render walls icons properly", () => {
    wrapper = shallow(<LMGridObjectIcon tag={WALL_D} />);
    expect(wrapper.contains("border_clear")).toBe(true);
  });
});
