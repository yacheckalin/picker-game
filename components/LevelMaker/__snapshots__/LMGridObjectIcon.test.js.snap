// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LMGridObjectIcon /> renders and displays properly 1`] = `
<styled.i
  className="material-icons text-darken-2 red-text  "
>
  lock_outline
</styled.i>
`;
